pageserver/src/tenant/timeline.rs

```diff
@@ -4428,26 +4438,27 @@ impl Timeline {
                     lsn
                 },
             );
-            Ok(ImageLayerCreationOutcome {
-                unfinished_image_layer: Some(image_layer_writer),
-                next_start_key: img_range.end,
+            Ok(ImageLayerCreationOutcome::Generated {
+                unfinished_image_layer: image_layer_writer,
             })
         } else {
             // Special case: the image layer may be empty if this is a sharded tenant and the
             // partition does not cover any keys owned by this shard.  In this case, to ensure
             // we don't leave gaps between image layers, leave `start` where it is, so that the next
```

Comment:
nit: comment needs updating, we removed `start`.

Reply:
Fixed in https://github.com/neondatabase/neon/pull/10572/commits/5f03e9d933a94a8d86b555d3385c4d919416a03f